### PR TITLE
LVNOD-80 Added guest controller, routes, modified user.create

### PR DESCRIPTION
**Pull request by AlbaMontis**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-80)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-116)

## Code reviewers

- [ ] @yarvanya
- [ ] @denisbodnar
- [ ] @shyyka

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

### CODE REVIEW

**Review by yarvanya — DISMISSED** [review]

File: server/helper/passwordGenerator.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+'use strict'
+
+module.exports = {
+  passwordGenerate() {
+    const chars =
+      ['qwertyuiopasdfghjklzxcvbnm', 'QWERTYUIOPASDFGHJKLZXCVBNM', '$@$!%*?&', '1234567890'];
+    let password = '';
+    let i;
+
+    for (i = 0; i < 3; i++) {
```
Comment (4asof4ik) [inline server/helper/passwordGenerator.js]: let i = 0 maybe?

**Review by denisbodnar — DISMISSED** [review]

File: server/controllers/guest.controller.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+module.exports = {
+
+  retrieve(req, res) {
+
+    Guest.findAll({
+      where: {event_id: req.params.id},
+      include: [User]
+    })
+    .then(guests => {
+      res.send({guest: guests});
```
Comment (4asof4ik) [inline server/controllers/guest.controller.js]: guests: guests

File: server/controllers/guest.controller.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+const Guest = require('../../config/db').Guest;
+const User = require('../../config/db').User;
+const Event = require('../../config/db').Event;
+const Profile = require('../../config/db').Profile;
+const mailer = require('./../helper/mailer');
+const URL = require('./../helper/constants');
+
+module.exports = {
+
+  retrieve(req, res) {
```
Comment (4asof4ik) [inline server/controllers/guest.controller.js]: list

File: server/controllers/guest.controller.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        event_id: eventId
+      },
+      include: [{
+        model: User,
+        include: [Profile]
+      }, {model: Event}]
+    })
+    .then(guests => {
+      guests.map(guest => {
+        const firstname = guest.User.Profile ? guest.User.Profile.first_name : '';
```
Comment (4asof4ik) [inline server/controllers/guest.controller.js]: guest.User.Profile.first_name || ''

File: server/controllers/guest.controller.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+          email: guest.User.email,
+          eventName: guest.Event.name,
+          date: guest.Event.date_event,
+          eventDescription: guest.Event.description,
+          img: 'party.jpg'
+        }, template);
+      });
+      return guests;
+    })
+    .then((guests) => res.send({guests: guests}))
```
Comment (4asof4ik) [inline server/controllers/guest.controller.js]: if one arg  we dont need ()

File: server/controllers/user.controller.js [hunk 6]
```diff
@@ -2,14 +2,45 @@
...
+const password = require('./../helper/passwordGenerator');
 
 module.exports = {
   create(req, res) {
     let assignUser = Object.assign({}, req.body);
-
-    User.create(assignUser)
-    .then(user => res.status(201).send(user))
-    .catch(error => res.status(400).send(error));
+    const id = req.body.eventId;
```
Comment (4asof4ik) [inline server/controllers/user.controller.js]: eventID or somethink like that

File: server/controllers/user.controller.js [hunk 7]
```diff
@@ -2,14 +2,45 @@
...
+    id ?
+      req.body.emails.map(email => {
+        User.findOne({
+          where: {
+            email: email
+          }
+        })
+        .then(user => user ?
```
Comment (4asof4ik) [inline server/controllers/user.controller.js]: use && and ||

File: server/controllers/user.controller.js [hunk 8]
```diff
@@ -2,14 +2,45 @@
...
+            event_id: id,
+            user_id: user.id
+          }) :
+          User.create({
+            email: email,
+            password: password.passwordGenerate(),
+            is_invited: true
+          })
+          .then(user => {
+            Guest.create({
```
Comment (4asof4ik) [inline server/controllers/user.controller.js]: you have 20-23

File: server/controllers/guest.controller.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    .catch(err => res.status(404).send(error));
```
Comment (4asof4ik) [inline server/controllers/guest.controller.js]: 400

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: server/controllers/guest.controller.js (hunk 3 above)
Comment (AlbaMontis) [reply server/controllers/guest.controller.js]: Done.

File: server/controllers/guest.controller.js (hunk 2 above)
Comment (AlbaMontis) [reply server/controllers/guest.controller.js]: Done.

File: server/controllers/guest.controller.js (hunk 4 above)
Comment (AlbaMontis) [reply server/controllers/guest.controller.js]: Done.

File: server/controllers/guest.controller.js (hunk 5 above)
Comment (AlbaMontis) [reply server/controllers/guest.controller.js]: Done.

File: server/controllers/guest.controller.js (hunk 9 above)
Comment (AlbaMontis) [reply server/controllers/guest.controller.js]: Done.

File: server/controllers/user.controller.js (hunk 6 above)
Comment (AlbaMontis) [reply server/controllers/user.controller.js]: Done.

File: server/helper/passwordGenerator.js (hunk 1 above)
Comment (AlbaMontis) [reply server/helper/passwordGenerator.js]: Done.

**Review by denisbodnar — APPROVED** [review]

**Review by yarvanya — APPROVED** [review]

**Review by 4asof4ik — APPROVED** [review]

**Pull request merged**